cpu/ppc64_cpu_test.py

```diff
@@ -67,6 +69,49 @@ def setUp(self):
         self.value = ""
         self.max_smt_value = int(self.curr_smt)
 
+    def test_build_upstream(self):
+        """
+        For upstream target download and compile source code
+        Caution : This function will overwrite system installed
+        lsvpd Tool binaries with upstream code.
+        """
+        if self.run_type == 'upstream':
+            self.detected_distro = distro.detect()
+            deps = ['gcc', 'make', 'automake', 'autoconf', 'bison', 'flex',
+                    'libtool', 'zlib-devel', 'ncurses-devel', 'librtas-devel']
+            if 'SuSE' in self.detected_distro.name:
+                deps.extend(['libnuma-devel'])
+            elif self.detected_distro.name in ['centos', 'fedora', 'rhel']:
+                deps.extend(['numactl-devel'])
+            else:
+                self.cancel("Unsupported Linux distribution")
+            for package in deps:
+                if not self.sm.check_installed(package) and not \
+                        self.sm.install(package):
+                    self.cancel("Fail to install %s required for this test." %
+                                package)
+            url = self.params.get(
+                'ppcutils_url', default='https://github.com/'
+                'ibm-power-utilities/powerpc-utils/archive/refs/heads/'
+                'master.zip')
+            tarball = self.fetch_asset('ppcutils.zip', locations=[url],
+                                       expire='7d')
+            archive.extract(tarball, self.workdir)
+            self.sourcedir = os.path.join(self.workdir, 'powerpc-utils-master')
+            os.chdir(self.sourcedir)
+            cmd_result = process.run('./autogen.sh', ignore_status=True,
+                                     sudo=True, shell=True)
+            if cmd_result.exit_status:
+                self.fail('Upstream build: Pre configure step failed')
+            cmd_result = process.run('./configure --prefix=/usr',
+                                     ignore_status=True, sudo=True, shell=True)
+            if cmd_result.exit_status:
+                self.fail('Upstream build: Configure step failed')
+            build.make(self.sourcedir)
+            build.make(self.sourcedir, extra_args='install')
+        else:
+            self.cancel("This test is supported with upstream as target")
```

Comment:
if default is distro, than will it cancel ?

Reply:
yes.  I have already provided output of test run with distro as a target